group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/OptimizedUniformAssignmentBuilderTest.java

```diff
@@ -557,10 +612,52 @@ public void testReassignmentWhenOneSubscriptionRemovedAfterInitialAssignmentWith
             mkTopicAssignment(topic2Uuid, 1)
         ));
 
+        GroupSpecImpl groupSpec = new GroupSpecImpl(
+            members,
+            HOMOGENEOUS,
+            partitionAssignments(members)
+        );
+        SubscribedTopicMetadata subscribedTopicMetadata = new SubscribedTopicMetadata(topicMetadata);
+
+        GroupAssignment computedAssignment = assignor.assign(
+            groupSpec,
+            subscribedTopicMetadata
+        );
+
         assertAssignment(expectedAssignment, computedAssignment);
         checkValidityAndBalance(members, computedAssignment);
     }
 
+    /**
+     * Generate a map of partition assignments from the given member spec.
+     *
+     * @param memberSpec        A map where the key is the member Id and the value is an
+     *                          AssignmentMemberSpec object containing the member's partition assignments.
+     * @return Map of topic partition to member assignments.
+     */
+    public Map<Uuid, Map<Integer, String>> partitionAssignments(
```

Comment:
This seems to be a copy from the general uniform assignment test. Could we share it? We could perhaps move it to a utility test class (e.g. AssignorTestUtils).